set.mm

```diff
@@ -659185,6 +659196,173 @@ D Fn ( I ... ( M - 1 ) ) /\
+    odm1inv.g $e |- ( ph -> G e. Grp ) $.
+    odm1inv.1 $e |- ( ph -> A e. X ) $.
+    $( The (order-1)th multiple of an element is its inverse.  (Contributed by
+       SN, 31-Jan-2025.) $)
+    odm1inv $p |- ( ph -> ( ( ( O ` A ) - 1 ) .x. A ) = ( I ` A ) ) $=
+      ( cfv co c1 wcel wceq eqid syl csg c0g cmin odid mulg1 oveq12d cgrp odcld
+      cz nn0zd 1zzd mulgsubdir syl13anc grpinvval2 syl2anc 3eqtr4d ) ABFNZBCOZP
+      BCOZDUANZOZDUBNZBUTOZUQPUCOBCOZBENZAURVBUSBUTABGQZURVBRMBCDFGVBHIJVBSZUDT
+      AVFUSBRMGCDBHJUETUFADUGQZUQUIQPUIQVFVDVARLAUQABGDFHIMUHUJAUKMGCDUQUTPBHJU
+      TSZULUMAVHVFVEVCRLMGDUTEBVBHVIKVGUNUOUP $.
+  $}
+
+  ${
+    $d S a $.  $d G a $.  $d ph a $.
+    finodsubmsubg.o $e |- O = ( od ` G ) $.
+    finodsubmsubg.g $e |- ( ph -> G e. Grp ) $.
+    finodsubmsubg.s $e |- ( ph -> S e. ( SubMnd ` G ) ) $.
+    finodsubmsubg.1 $e |- ( ph -> A. a e. S ( O ` a ) e. NN ) $.
+    $( A submonoid whose elements have finite order is a subgroup.
+       (Contributed by SN, 31-Jan-2025.) $)
+    finodsubmsubg $p |- ( ph -> S e. ( SubGrp ` G ) ) $=
+      ( cfv wcel wral wa co cmg wceq cbs eqid syl ad2antrr csubmnd cminusg cmin
+      csubg cv cn c1 cgrp adantr wss submss sselda odm1inv cmnd submmnd nnm1nn0
+      cress adantl ressbas2 eleqtrd mulgnn0cl syl3anc submmulg 3eltr4d eqeltrrd
+      cn0 simplr ex ralimdva mpd wb issubg3 mpbir2and ) ABCUDJKZBCUAJKZEUEZCUBJ
+      ZJZBKZEBLZHAVPDJZUFKZEBLVTIAWBVSEBAVPBKZMZWBVSWDWBMZWAUGUCNZVPCOJZNZVRBWD
+      WHVRPWBWDVPWGCVQDCQJZWIRZFWGRZVQRZACUHKZWCGUIABWIVPAVOBWIUJZHWIBCWJUKSZUL
+      UMUIWEWFVPCBUQNZOJZNZWPQJZWHBWEWPUNKZWFVFKZVPWSKWRWSKAWTWCWBAVOWTHBWPCWPR
+      ZUOSTWBXAWDWAUPURZWEVPBWSAWCWBVGZABWSPZWCWBAWNXEWOBWIWPCXBWJUSSTZUTWSWQWP
+      WFVPWSRWQRZVAVBWEVOXAWCWHWRPAVOWCWBHTXCXDBWGWQCWPWFVPWKXBXGVCVBXFVDVEVHVI
+      VJAWMVNVOVTMVKGEBCVQWLVLSVM $.
+  $}
+
+  ${
+    $d .0. a $.  $d G a $.
+    sn-0subg.z $e |- .0. = ( 0g ` G ) $.
+    $( A shorter proof of ~ 0subg using ~ df-od .  (Contributed by SN,
+       31-Jan-2025.) $)
+    sn-0subg $p |- ( G e. Grp -> { .0. } e. ( SubGrp ` G ) ) $=
+      ( va cgrp wcel csn cod cfv eqid id cmnd csubmnd grpmnd 0subm syl cn cv c1
+      wral od1 1nn eqeltrdi fvexi wceq fveq2 eleq1d ralsn sylibr finodsubmsubg
+      c0g ) AEFZBGZAAHIZDUNJZULKULALFUMAMIFANABCOPULBUNIZQFZDRZUNIZQFZDUMTULUPS
+      QAUNBUOCUAUBUCUTUQDBBAUKCUDURBUEUSUPQURBUNUFUGUHUIUJ $.
+  $}
+
+  ${
+    $d S a $.  $d G a $.  $d ph a $.
+    finsubmsubg.b $e |- B = ( Base ` G ) $.
+    finsubmsubg.g $e |- ( ph -> G e. Grp ) $.
+    finsubmsubg.s $e |- ( ph -> S e. ( SubMnd ` G ) ) $.
+    finsubmsubg.1 $e |- ( ph -> B e. Fin ) $.
+    $( A submonoid of a finite group is a subgroup.  This does not extend to
+       infinite groups, even if we include the inverses of the elements of the
+       submonoid; for example, ((the submonoid ` S = A + B sqrt 2 ` where
+       ` A , B e. NN ` ) together with their negations) only contains elements
```

Comment:
Write that you're talking about the submonoid **of the group of reals for addition**.

You need $0 \in S$, so maybe the most natural thing to do is to write "` A , B e. NN0 `" and then "with absolute value at least 1".